File: Antony-Ndungu/Arrays and Strings 1/remove-element.go
package main

import "fmt"

func main() {
	nums := []int{3, 2, 2, 3}
	nums = nums[0:removeElement(nums, 3)]
	fmt.Println(nums)
}

func removeElement(nums []int, val int) int {
	var i int
	for i < len(nums) {
		if val == nums[i] {
			nums[i] = nums[len(nums)-1]
			nums = nums[:len(nums)-1]
			continue
		}
		i++
	}
	return len(nums)
}

File: Antony-Ndungu/Arrays and Strings 1/remove-element.c
#include <stdio.h>

int removeElement(int* nums, int numsSize, int val);

int main(void){
    int nums[] = {3,2,2,3};
    int numsSize = (int) sizeof(nums)/sizeof(int); 
    numsSize =  removeElement(nums, numsSize, 3);
    printf("size of nums: %d\n",numsSize);
    printf("elements in nums: ");
    for(size_t i = 0; i < numsSize; i++)
        printf("%d ", nums[i]);
    printf("\n");
    return 0;
}

int removeElement(int* nums, int numsSize, int val){
    size_t i = 0;
    while(i<numsSize){
        if(val == nums[i]){
            nums[i] = nums[numsSize - 1];
            numsSize--;
            continue;
        }
        i++;
    }
}